logging/src/main/java/org/jboss/as/logging/capabilities/LoggingProfileCapabilityRecorder.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.logging.capabilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.as.controller.CapabilityReferenceRecorder;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.logging.CommonAttributes;
+
+/**
+ * Allows the name of the logging profile to be appended to the capability name.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+class LoggingProfileCapabilityRecorder implements CapabilityReferenceRecorder {
```

Comment:
@jamezp maybe the logging code could benefit from https://github.com/wildfly/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/CapabilityReferenceRecorder.java#L325 with the right mapper?

Reply:
Unfortunately I don't think it will work. The dynamic name is always based on the resource address so the value of the requirement gets the name from the address rather than the attribute.